<a href="https://colab.research.google.com/github/MariKa666/repository-test-1/blob/main/MY_hw_9_1_%D0%9E%D0%BF%D0%B5%D1%80%D0%B0%D1%86%D1%96%D1%97_%D0%BD%D0%B0%D0%B4_%D0%B4%D0%B0%D0%BD%D0%B8%D0%BC%D0%B8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [2]:
a = 101
b = 210
c = 111
d = 402
rez = (a + b)*c/d
print(rez)


85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [3]:
11283%89

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [4]:
11283//89

126

4. Запишіть у змінну `name` ваше імʼя.

In [6]:
name = "Maryna"

5. Запишіть у змінну `surname` ваше прізвище.

In [7]:
surname = "Katseba"

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [9]:
name_surname = name + " " + surname
print(name_surname)

Maryna Katseba


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [10]:
print(name_surname.upper())

MARYNA KATSEBA


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [11]:
print(name_surname[2])

r


Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [12]:
print(name_surname[0], name_surname[-1])

M a


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [13]:
len(name_surname)

14

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [15]:
capital = "Antananarivo"
capital_a = capital.lower()
count_a = capital_a.count('a')
print(count_a)

4


Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [18]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
print(my_string.startswith("The"))
my_string.endswith("dog!")

True


False

13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [21]:
my_string.split( )
my_string.join(_)
print(my_string)

# а може простіше?
my_string.replace(" ", "_")

The quick brown fox jumps over the lazy dog!!!


'The_quick_brown_fox_jumps_over_the_lazy_dog!!!'

Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`